src/UI/Buyer/src/app/products/containers/product-list/product-list.component.ts

```diff
@@ -94,6 +97,7 @@ export class ProductListComponent implements OnInit {
 
   changeCategory(category: string): void {
     this.addQueryParam({ category });
+    if (this.isModalOpen) this.closeCategoryModal()
```

Comment:
It seems odd to have a boolean just to tell if the modal is open. What happens if you just close the modal, and its already closed?

If we do have to keep the boolean, can you wrap `this.closeCategoryModal()` in brackets? Makes it a little harder to accidentally break

Reply:
do you mean something like this?  if (this.isModalOpen) {this.closeCategoryModal()}?

Reply:
yiss

Reply:
i should probably add it to tslint. i thought it was in there once upon a time